incubator/binding-mqtt-kafka/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/kafka/internal/stream/MqttKafkaSessionFactory.java

```diff
@@ -2946,7 +3169,7 @@ private void onKafkaReset(
             assert acknowledge <= sequence;
             assert acknowledge >= delegate.initialAck;
 
-            delegate.initialAck = acknowledge;
+            delegate.initialAck = acknowledge - delegate.sessionPadding;
```

Comment:
Same feedback as above.

Reply:
When we receive a reset, we set the delegate.initialAck to the reset frame ack, which is the greater then what mqtt sent. E.g: we don't have will message in the mqtt begin frame, so we only send the expiry cancellation + expire later signal, which is 32 bytes. This will be acked by Kafka, so onKafkaReset the resetl.acknowledge is 32. But mqtt didn't send any data, so delegate.initialSeq = 0. So maybe in reality, we'd need to remove the "delegate.initialAck = acknowledge;" assignment?

Reply:
Yeah, I think this raises a more significant point about the relationship between the `mqtt` session stream and the multiple `kafka` streams that see the same `mqtt` session stream as their `delegate`.

In general, each stream should be managing its own `initialSeq`, `initialAck`, etc locally to the stream.

In special cases, say when we are proxying 1-to-1 between streams, then it is feasible to track `initialSeq`, `initalAck`, etc on one stream only and use the same values for both.

However, in this case, it seems like with the 1-to-many relationship from `mqtt` session stream to `kafka` streams, we are no longer proxying in the special case, so we are trying to handle the divergence on a case by case basis.

I realize the design has evolved a bit during implementation. Perhaps we want to take one of those `kafka` streams and treat them as the primary, similar to your suggestion of not updating the delegate in the non-primary `kafka` stream.  If that is sufficient, then we still need to track `initialSeq`, `initialAck`, etc separately in the non-primary `kafka` stream as it will likle yneed to use its own values for sending its frames.

Let's perhaps also rethink the naming, so that a stream is not named as a `...Proxy` if it is not behaving as a proxy. In some cases, perhaps we should be naming such classes `...Stream` instead. There may be other good alternative name suffixes too, that might more clearly help to convey the intent of the design.